tests/test_cuda_bd_exchange_mover.py

```diff
@@ -330,35 +325,122 @@ def test_bd_exchange_deterministic_moves(proposals_per_move, precision, seed):
+        1,
+        batch_size=batch_size,
+    )
 
     iterative_moved_coords = conf.copy()
     for _ in range(proposals_per_move):
         iterative_moved_coords, _ = bdem_a.move(iterative_moved_coords, box)
-        assert not np.all(conf == iterative_moved_coords)  # We should move every time since its a single mol
+        assert not np.all(conf == iterative_moved_coords)
     batch_moved_coords, _ = bdem_b.move(conf, box)
+
+    assert bdem_a.n_accepted() > 0
+    assert bdem_a.n_proposed() == proposals_per_move
+    assert bdem_a.n_accepted() == bdem_b.n_accepted()
+    assert bdem_a.n_proposed() == bdem_b.n_proposed()
+
     # Moves should be deterministic regardless the number of steps taken per move
     np.testing.assert_array_equal(iterative_moved_coords, batch_moved_coords)
 
+
+@pytest.mark.parametrize("proposals_per_move, batch_size", [(2, 2), (100, 100)])
+@pytest.mark.parametrize("precision", [np.float64, np.float32])
+@pytest.mark.parametrize("seed", [2024])
+def test_bd_exchange_deterministic_batch_moves(proposals_per_move, batch_size, precision, seed):
+    """Verify that if we run with the same batch size but either call `move()` repeatedly or just
+    increase the number of proposals per move in the constructor that the results should be identical
+    """
+    ff = Forcefield.load_default()
+    system, conf, _, _ = builders.build_water_system(1.0, ff.water_ff)
+    bps, _ = openmm_deserializer.deserialize_system(system, cutoff=1.2)
+
+    nb = next(bp for bp in bps if isinstance(bp.potential, Nonbonded))
+    bond_pot = next(bp for bp in bps if isinstance(bp.potential, HarmonicBond)).potential
+
+    group_idxs = get_group_indices(get_bond_list(bond_pot), conf.shape[0])
+
+    rng = np.random.default_rng(seed)
+
+    box = np.eye(3) * 100.0
+
+    N = conf.shape[0]
+
+    params = nb.params
+
+    cutoff = nb.potential.cutoff
+    klass = custom_ops.BDExchangeMove_f32
+    if precision == np.float64:
+        klass = custom_ops.BDExchangeMove_f64
+
+    iterations = rng.integers(2, 5)
+
+    # Reference that makes proposals_per_move proposals per move() call
+    bdem_a = klass(
+        N,
+        group_idxs,
+        params,
+        DEFAULT_TEMP,
+        nb.potential.beta,
+        cutoff,
+        seed,
+        proposals_per_move,
+        1,
+        batch_size=batch_size,
+    )
+    # Test version that makes all proposals in a single move() call
+    bdem_b = klass(
+        N,
+        group_idxs,
+        params,
+        DEFAULT_TEMP,
+        nb.potential.beta,
+        cutoff,
+        seed,
+        proposals_per_move * iterations,
+        1,
+        batch_size=batch_size,
+    )
+
+    iterative_moved_coords = conf.copy()
+    for _ in range(iterations):
+        iterative_moved_coords, _ = bdem_a.move(iterative_moved_coords, box)
+        assert not np.all(conf == iterative_moved_coords)
+    batch_moved_coords, _ = bdem_b.move(conf, box)
+
     assert bdem_a.n_accepted() > 0
```

Comment:
document what a typical `n_accepted()` is?

Reply:
Added and increased the acceptance requirement to half the moves: https://github.com/proteneer/timemachine/pull/1274/commits/0c1cdf823244d222f2e7fb0de475bf58ce85efab